homeassistant/components/intent_script/__init__.py

```diff
@@ -55,10 +62,27 @@
 )
 
 
-async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
-    """Set up the intent script component."""
-    intents = config[DOMAIN]
+async def async_reload(hass: HomeAssistant, servie_call: ServiceCall) -> None:
+    """Handle start Intent Script service call."""
+    new_config = await async_integration_yaml_config(hass, DOMAIN)
+    existing_intents = hass.data[DOMAIN]
+
+    for intent_type in existing_intents:
+        intent.async_remove(hass, intent_type)
+
+    if not new_config or DOMAIN not in new_config:
+        hass.data[DOMAIN] = {}
+        return
+
+    new_intents = new_config[DOMAIN]
+
+    async_load_intents(hass, new_intents)
+
+
+def async_load_intents(hass: HomeAssistant, intents: dict):
```

Comment:
Missing return value typing. Please type the whole signature when adding type annotations.